### PR TITLE
add setting to disable automatic smartling lock id generation (WP-613)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

File: build/assets/js/smartling-locked.js [hunk 1]
```diff
@@ -47,7 +31,7 @@ const sidebarWithLockingCheckbox = createHigherOrderComponent((BlockEdit) => {
                                 newAttributes[LOCKED] = value;
                                 props.setAttributes(newAttributes);
                             }}
-                            help={props.attributes[LOCKED] ? 'Content will not change on translation' : 'Content will change on translation'}
+                            help={props.attributes[LOCKED] ? __('Content will not change on translation') : __('Content will change on translation')}
```
Comment (PavelLoparev) [inline build/assets/js/smartling-locked.js:34]: Is `__()` used for UI strings translation?

File: inc/Smartling/Bootstrap.php [hunk 2]
```diff
@@ -225,6 +225,7 @@ public function updateGlobalExpertSettings () {
         GlobalSettingsManager::setPageSize($pageSize);
         GlobalSettingsManager::setLoggingCustomization($data['loggingCustomization']);
         GlobalSettingsManager::setHandleRelationsManually((int)$data['handleRelationsManually']);
+        GlobalSettingsManager::setGenerateLockIdsFrontend($data['handleRelationsManually'] === "1");
```
Comment (PavelLoparev) [inline inc/Smartling/Bootstrap.php:228]: Why we enable automatic block id generation only if `handleRelationsManually` is enabled?

File: inc/Smartling/Services/GlobalSettingsManager.php [hunk 3]
```diff
@@ -251,6 +251,27 @@ public static function setHandleRelationsManually($value)
         }
     }
 
+    public const SMARTLING_FRONTEND_GENERATE_LOCK_IDS = 'smartling_frontend_generate_lock_ids';
+
+    public static function isGenerateLockIdsFrontendDefault(): bool
+    {
+        return false;
+    }
```
Comment (PavelLoparev) [inline inc/Smartling/Services/GlobalSettingsManager.php:259]: Should be constant instead, I think

File: inc/Smartling/WP/Controller/PostBasedWidgetControllerStd.php [hunk 4]
```diff
@@ -440,13 +441,15 @@ public function preView($post)
...
                         ]
                     );
-                    wp_enqueue_script(
-                        'smartling-block-locking',
-                        $this->getPluginInfo()->getUrl() . 'js/smartling-block-locking.js',
-                        ['wp-blocks', 'wp-i18n', 'wp-element', 'wp-editor'],
-                        $this->getPluginInfo()->getVersion(),
-                        true,
-                    );
+                    if (GlobalSettingsManager::isGenerateLockIdsFrontend()) {
```
Comment (PavelLoparev) [inline inc/Smartling/WP/Controller/PostBasedWidgetControllerStd.php:444]: It doesn't sound right. I would prefer to have name like `isGenerateLockIdsEnabled`.

File: inc/Smartling/WP/View/ConfigurationProfiles.php [hunk 5]
```diff
@@ -201,6 +201,28 @@ class="toggleExpert"><strong><?= __('Show Expert Settings'); ?></strong></a>
                             <a href="javascript:void(0)" id="resetPageSize" data-default="<?= GlobalSettingsManager::getPageSizeDefault(); ?>">reset to defaults</a>
                         </td>
                     </tr>
+                    <tr>
+                        <th><label for="<?= GlobalSettingsManager::SMARTLING_FRONTEND_GENERATE_LOCK_IDS ?>">Generate lock ids on editing content</label></th>
```
Comment (PavelLoparev) [inline inc/Smartling/WP/View/ConfigurationProfiles.php:205]: Should this string be wrapped to `__()`?

**Review by PavelLoparev — CHANGES_REQUESTED** [review]

File: build/assets/js/smartling-locked.js (hunk 1 above)
Comment (vsolovei-smartling) [reply build/assets/js/smartling-locked.js:34]: Yes

File: inc/Smartling/Bootstrap.php (hunk 2 above)
Comment (vsolovei-smartling) [reply inc/Smartling/Bootstrap.php:228]: Fixed

File: inc/Smartling/Services/GlobalSettingsManager.php (hunk 3 above)
Comment (vsolovei-smartling) [reply inc/Smartling/Services/GlobalSettingsManager.php:259]: It serves no purpose, I've inlined all similar constants for consistency

File: inc/Smartling/WP/Controller/PostBasedWidgetControllerStd.php (hunk 4 above)
Comment (vsolovei-smartling) [reply inc/Smartling/WP/Controller/PostBasedWidgetControllerStd.php:444]: Done

File: inc/Smartling/WP/View/ConfigurationProfiles.php (hunk 5 above)
Comment (vsolovei-smartling) [reply inc/Smartling/WP/View/ConfigurationProfiles.php:205]: Done

**Review by PavelLoparev — APPROVED** [review]

**Pull request merged**